hazelcast/src/main/java/com/hazelcast/spi/impl/AbstractCompletableFuture.java

```diff
@@ -161,17 +181,23 @@ public boolean isCancelled() {
     }
 
     @Override
-    public V get() throws InterruptedException, ExecutionException {
-        try {
-            return get(Long.MAX_VALUE, TimeUnit.MILLISECONDS);
-        } catch (TimeoutException e) {
-            logger.severe("Unexpected timeout while processing " + this, e);
-            return null;
+    public final V get() throws InterruptedException, ExecutionException {
+        for (; ; ) {
+            try {
+                return get(Long.MAX_VALUE, TimeUnit.MILLISECONDS);
+            } catch (TimeoutException e) {
+            }
```

Comment:
To forestall future raised eyebrows and suspicion, maybe add a comment here, such as: 
```
 // A timeout here can only be a spurious artifact.
 // It should never happen and even if it does, we must retry.
```